src/md_compress_adaptive_adios2.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+import math
+
+from pprint import pprint
+from codar.oas.MDTrSampler import MDTrSampler
+import codar.utils.pyAdios as AD
+
+from mpi4py import MPI
+
+comm = MPI.COMM_WORLD
+rank = comm.Get_rank()
+size = comm.Get_size()
+
+# For SST ending, replace 'BP' with 'SST'.
+reader = AD.pyAdios('BP')
+reader.open('../src/nwchem_xyz.bp', AD.OpenMode.READ)
+
+# to get available attributes
+#attrs = reader.available_attributes()
+#pprint(attrs)
+
+# to get available variables
+avars = reader.available_variables()
+if rank == 0:
+    print('Available Variables:')
+    pprint(avars)
+
+# Q: in the variables, there are 'NATOMS', 'OFFSET' and 'TATOMS'.
+#    is one of them related to natoms? It seems 'TATOMS' matches with the
+#    'ntoms', i.e. the length of 1D array.
+natoms = avars['LX'].shape[0]
+
+# compute the number of atoms and offset for each rank.
+# (special care for the last rank)
+localatoms = math.ceil(natoms/size)
+if rank == (size - 1):
+    localatoms = natoms - localatoms * (size - 1)
+offset = math.ceil(natoms/size) * rank
+
+# create sampler object
+mds2 = None
+if rank == 0:
+    mds2 = MDTrSampler(natoms, 2, conv_size=10, n_samples=32, batch_size=50, manifold_size=25)
+
+# maybe not necessary, just to make sure all ranks get the file (stream) handler
+# but, this could block all MPI processors forever, if one of processor fails to get
+# stream in SST engine.
+comm.Barrier()
+
+# loop over steps
+step = reader.current_step()
+while step >= 0:
```

Comment:
@bsmind 

Let's also ask how to terminate this loop... 

Shall we sync with Huub on how to send the terminal condition or out of stream as terminal?